### ISSUE TITLE
Fixing LetsEncrypt auto renewal

### ISSUE THREAD

**Issue by cbarratt** [Ubiquiti-App/UNMS#92]
I'm running 0.10.3 and I can't see any options in UNMS RE: SSL certificate config (I feel like it used to be in the settings?).

I've been running UNMS fine now with a certificate, though today it has expired and I can't figure out a way to fix it and force it to renew the certificate. I'm a little stuck on the best course of action to resolve this :(

**Comment by jaroklima** [comment]
If you used the `--cert-*` installation arguments to supply your own SSL certificate, you will have to renew it manually. 

If your certificate was generated by UNMS, it should renew automatically before it expires. If this is the case, please check `~unms/data/logs/unms.*.log` for related error messages at around 4AM (every day) or send me the whole log file.

**Comment by cbarratt** [comment]
Hey @jaroklima - I tried to find any errors related to this but couldn't see any. Here's a gist of my UNMS log from today that includes messages regarding LetsEncrypt but I couldn't quite pinpoint what went wrong :(

here's a gist of the messages + errors from 4am

https://gist.github.com/cbarratt/9ada1163f228b30a40a471806a9dabd4

**Comment by jaroklima** [comment]
This is very interesting. According to the log, UNMS realizes that the certificate is due for renewal, calls the LetsEncrypt library to do the renewal, the library returns a success, UNMS reloads the certificate and yet it is not renewed. 

The certificate files are saved under ~unms/data/cert, could you please check the content of the directory and let me know what certificates are there and whether there are any files that were recently modified? I'm trying to figure out whether the LetsEncrypt library saves the new certificate and doesn't use it, or whether the files aren't being updated at all.

**Comment by cbarratt** [comment]
I’d already checked earlier today, it looks like the first original certs are still there from August 1st :( i can’t see any traces of a new certificate

**Comment by cbarratt** [comment]
So far it hasn't appeared to resolve itself. Is there any further debugging I can provide that may help you guys nail down this problem?

**Comment by rdg5589** [comment]
I'm experiencing this same issue with my UNMS server.  I see no evidence of new certs in the ~unms/data/certs directory.  The UNMS log does show the new cert being installed though.

**Comment by hhenne** [comment]
Had the same issue... check if you have any firewall rules or similar restricting incoming traffic on 80/443 to specific ips, if you do try disabling those temporarily, this seemed to fix it for me and the certs renewed ok.

**Comment by Ve2mrx** [comment]
Got bitten by this bug today, UNMS server triggers an "Insecure" warning in Firefox, and mentions HSTS. I can't get any cert info from Firefox, and in ~unms/data/cert my cert and key are dated December 18th 2017.

Logs have been captured, Support info also. I will analyse this when I get back  (I'm on the road).

**Comment by florisvdk** [comment]
@Ve2mrx in chrome you can bypas the hsts page by typing badidea , then you can go to the login page. Got bitten by that when setting up the system

**Comment by Ve2mrx** [comment]
I looked into it. I did many things, going by instinct (so undocumented) and managed to fix it. 
 What fixed it included:

- Setting the server clock to the right time (It was set one hour wrong yesterday, my bad!)
- Renaming the old cert files,
- Restarting UNMS,
- Reinstalling UNMS. 

I then logged in using Chrome, got a warning about the site being insecure. I just asked to "Let me in anyway".

In the settings, I had a note saying the Let's Encrypt certificate was disabled for too many failed attempts to renew. I forgot to take note of the mentioned text.

I turned the Let's Encrypt switch back ON, and Applied. It worked.

**Comment by sxpert** [comment]
here's what I get, which may be useful in tracking the bug
Let’s Encrypt was turned off because the previous attempt had failed.

Timestamp: Today at 6:03
Error: An unexpected error occurred:
ConnectionError: HTTPSConnectionPool(host='acme-v01.api.letsencrypt.org', port=443): Max retries exceeded with url: /directory (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f7919468950>: Failed to establish a new connection: [Errno -3] Try again',))
Please see the logfiles in /tmp for more details.

**Comment by peatrick** [comment]
I recently upgraded my ER-X to an ER12 and during that process _all_ machines on the network were assigned new internal IP addresses, as well as getting a new WAN IP assigned from my ISP.  This almost certainly caused the aforementioned bug.

The way i worked around this was by navigating to: `/home/unms/data/cert` and proceeding to delete all certificate (`rm *.crt`) as well as key files (`rm *.key`) and double check the "keys" folder to remove those files, just in case.  After doing this I rebooted UNMS, but wasn't able to log in due to the HSTS header issue in Chrome telling me it's not safe to log in.  Simply accessing the page via FQDN in an incognito browser seemed to easily work around this.  At this point I was able to use the GUI to refresh the Let's Encrypt certificate and I'm once again showing "Secure" when accessing UNMS, however it seemed to have preserve the original date I created the certificate, otherwise no issues i've noticed, at least initially:  https://i.imgur.com/ZbMxMpz.png

Obviously, YMMV.  Best of luck, thanks for all who contributed to this issue.
> **UNMS v0.12.2**
> **Debian GNU/Linux 9.5 (stretch)**
